cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/oauth2/OAuthClient.java

```diff
@@ -123,26 +120,35 @@ public TokenProvider getTokenProvider() {
         return new OAuthTokenProvider(this);
     }
 
-    @SuppressWarnings({ "rawtypes"})
+    @SuppressWarnings({ "rawtypes" })
     public void changePassword(String oldPassword, String newPassword) {
-        HttpHeaders headers = new HttpHeaders();
-        headers.add(AUTHORIZATION_HEADER_KEY, token.getTokenType() + " " + token.getValue());
-        HttpEntity info = new HttpEntity(headers);
-        ResponseEntity<String> response = restTemplate.exchange(authorizationUrl + "/userinfo", HttpMethod.GET, info, String.class);
-        Map<String, Object> responseMap = JsonUtil.convertJsonToMap(response.getBody());
+        Map<String, Object> responseMap = webClient.get()
+                                                   .uri(authorizationUrl + "/userinfo")
+                                                   .header(HttpHeaders.AUTHORIZATION, getAuthorizationHeaderValue())
```

Comment:
Should we need this header in the response?
I think the AUHTORIZATION header is send below in webClient.put()...header()